PWGJE/Tasks/jetSpectraESE.cxx

```diff
@@ -0,0 +1,143 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+// jet analysis framework with ESE (19/08/2024)
+//
+/// \author Joachim Hansen <joachim.hansen@cern.ch>
+//
+
+#include "Framework/ASoA.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/HistogramRegistry.h"
+
+#include "Common/Core/RecoDecay.h"
+#include "Common/Core/TrackSelection.h"
+#include "Common/Core/TrackSelectionDefaults.h"
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+
+#include "PWGJE/Core/FastJetUtilities.h"
+#include "PWGJE/Core/JetDerivedDataUtilities.h"
+#include "PWGJE/DataModel/Jet.h"
+
+#include "PWGCF/Flow/DataModel/FlowESE.h"
+#include "PWGCF/Flow/Core/FFitWeights.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+#include "Framework/runDataProcessing.h"
+
+using ColWqVecFT0C = soa::Join<aod::Collisions, aod::CentFT0Cs, aod::qVecFT0Cs, aod::qPercentileFT0Cs>;
+using JColwESE = soa::Join<aod::JCollisions, aod::CentFT0Cs, aod::qVecFT0Cs, aod::qPercentileFT0Cs>;
```

Comment:
just an additonal comment on this. In general since JCollisions is a (self contained) derived data format, its not foreseen to have it always joinable tables that are at their core joinable with the aod::Collisions table. For now if you prefer to keep this this way it is fine but it can happen that some development we do makes JCollisions and Collisions not joinable anymore. The safer option would be to have in your process function process(soa::Join<JetCollisions, aod::JCollisionPIs>::iterator const& collision, soa::Join<aod::Collisions, aod::CentFT0Cs, aod::qVecFT0Cs, aod::qPercentileFT0Cs> const& ) and then inside this function do auto originalCollision = collision.collision_as<soa::Join<aod::Collisions, aod::CentFT0Cs, aod::qVecFT0Cs, aod::qPercentileFT0Cs>>() and then get the qvector etc from originalCollision

Reply:
Thanks a lot! I will change it in a future commit (I don't think it's important to change now to make this pr).

Reply:
ok but just to let you know that your task might stop working at some point unexpectedly due to this

Reply:
since you are making other changes anyways you could consider incorporating it

Reply:
I have changed it now, since Victor requested me to relocate some of the code. So it will make the next commit. Thanks for the suggestions. :)